server/src/test/java/ru/practicum/shareit/booking/BookingServiceIntegrationTest.java

```diff
@@ -0,0 +1,460 @@
+package ru.practicum.shareit.booking;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.Query;
+import jakarta.persistence.TypedQuery;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import ru.practicum.shareit.booking.dto.BookingDto;
+import ru.practicum.shareit.booking.dto.NewBookingDto;
+import ru.practicum.shareit.booking.entity.Booking;
+import ru.practicum.shareit.booking.enums.States;
+import ru.practicum.shareit.booking.enums.Statuses;
+import ru.practicum.shareit.booking.service.BookingService;
+import ru.practicum.shareit.item.dto.ItemDto;
+import ru.practicum.shareit.user.dto.UserDto;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.notNullValue;
+
+
+@Transactional
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+@SpringBootTest(
+        properties = "spring.datasource.username=shareit",
```

Comment:
в папке test/resources есть appication.properties со строкой
spring.datasource.username=shareit

есть ли необходимость указывать?

Reply:
Можно еще такую аннотацию рассмотреть [AutoConfigureTestDatabase](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/test/autoconfigure/jdbc/AutoConfigureTestDatabase.html)
Annotation that can be applied to a test class to configure a test database to use instead of the application-defined (но это больше для информации, что если ее использовать, то в  тестовом application.properties datasource можно совсем не настраивать)